internal/charmstore/store_test.go

```diff
@@ -128,6 +128,36 @@ func (s *StoreSuite) checkAddCharm(c *gc.C, ch charm.Charm, addToES bool, url *r
 	c.Assert(errgo.Cause(err), gc.Equals, params.ErrDuplicateUpload)
 }
 
+// addRequiredCharms adds any charms required by the given
+// bundle that are not already in the store.
+func (s *StoreSuite) addRequiredCharms(c *gc.C, bundle charm.Bundle) {
```

Comment:
Unfortunately there's a problem if we want storetesting to depend on internal/charmstore
because the internal/charmstore tests are internal so we'll have a cyclic dependency.
We can make the internal/charmstore tests external which avoids this, but I think we
can do that later.